docs/advanced_topics/api/django-ninja.md

```diff
@@ -0,0 +1,230 @@
+        model = BlogPage
+        model_fields = [
+            "intro",
+        ]
+
+    @staticmethod
+    def resolve_authors(page: BlogPage, context) -> List[str]:
+        return [author.name for author in page.authors.all()]
+```
+
+This could also be done with the Pydantic `Field` function if the `BlogPage` class had a method to retrieve author names directly: `authors: List[str] = Field([], alias="get_author_names")`.
+
+### Rich text in the API
+
+Rich text fields in Wagtail use a specific internal format, described in [](../../extending/rich_text_internals). They can be added to the schema as `str`, but it’s often more useful for the API to provide a “display” representation, where references to pages and images are replaced with URLs.
+This can also be done with [Ninja resolvers](https://django-ninja.dev/guides/response/#resolvers). Here is an example with the `HomePageSchema`:
+
+```python
+from wagtail.rich_text import expand_db_html
+
+
+class HomePageSchema(BasePageSchema, ModelSchema):
+    content_type: Literal["homepage"]
+    body: str
+
+    class Config(BasePageSchema.Config):
+        model = HomePage
+
+    @staticmethod
+    def resolve_body(page: HomePage, context) -> str:
+        return expand_db_html(page.body)
+```
+
+Here, `body` is defined as a `str`, and the resolver uses the `expand_db_html` function to convert the internal representation to HTML.
+
+### Images in the API
+
+We can use a similar technique for images, combining resolvers and aliases to generate the data.
+We use the [`get_renditions()` method](image_renditions_multiple) to retrieve the formatted images, and a custom `RenditionSchema` to define their API representation.
+
+```python
+from wagtail.images.models import AbstractRendition
+
+
+class RenditionSchema(ModelSchema):
+    # We need to use the Field / alias API for properties
+    url: str = Field(None, alias="file.url")
+    alt: str = Field(None, alias="alt")
+
+    class Config:
+        model = AbstractRendition
+        model_fields = [
+            "width",
+            "height",
+        ]
+```
+
+On the `BlogPageSchema`, we define our image field as: `main_image: List[RenditionSchema] = []`. Then add the resolver for it:
+
+```python
+    @staticmethod
+    def resolve_main_image(page: BlogPage) -> List[AbstractRendition]:
+        filters = [
+            "fill-800x600|format-webp",
+            "fill-800x600",
+        ]
+        if image := page.main_image():
+            return image.get_renditions(*filters).values()
+        return []
+```
+
+In JSON, our `main_image` is now represented as an array, where each item is an object with `url`, `alt`, `width`, and `height` properties.
+
+## OpenAPI documentation
+
+Django Ninja generates [OpenAPI](https://swagger.io/specification/) documentation automatically, based on the defined operations and schemas.
+This also includes a documentation viewer, with support to try out the API directly from the browser.
```

Comment:
Maybe add a suggestion to check it out?

```suggestion
This also includes a documentation viewer, with support to try out the API directly from the browser. With the above example, you can try accessing the docs at `/api/docs`.
```

(Annoyingly, the path doesn't have a trailing slash and adding it results in 404 in my case for some reason.)

Reply:
Let’s. This is already an early step in the mini tutorial at the top of the page but certainly worth mentioning again.